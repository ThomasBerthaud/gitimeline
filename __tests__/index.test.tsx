import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import { GetServerSidePropsContext } from 'next';
import { userRepositories } from './mocks/handlers';
import Home from '../pages';

describe('index page', () => {
  test('should display home page', () => {
    const { asFragment } = render(<Home />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('it should search for a repository');

  test('it should search for a user');
});
