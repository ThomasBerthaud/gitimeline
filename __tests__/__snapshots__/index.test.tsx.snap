// Vitest Snapshot v1

exports[`index page > should display home page 1`] = `
<DocumentFragment>
  <h1
    class="font-bold text-3xl text-center my-20"
  >
    Search for a Github user or repository
  </h1>
  <div
    class="flex justify-center h-full"
  >
    <section
      class="_section_diuk4_1 _left_diuk4_9"
    >
      <label
        class="self-start"
        for="value-Username"
      >
        Username
      </label>
      <input
        class="w-full mb-6"
        id="value-Username"
        name="value-Username"
        type="text"
      />
      <button
        class="w-full"
      >
        Search User
      </button>
    </section>
    <div
      class="relative border"
      style="width: 1px;"
    >
      <div
        class="absolute top-1/2 left-0 -translate-x-1/2 bg-white text-xl font-bold"
      >
        Or
      </div>
    </div>
    <section
      class="_section_diuk4_1 _right_diuk4_17"
    >
      <label
        class="self-start"
        for="value-Repository"
      >
        Repository
      </label>
      <input
        class="w-full mb-6"
        id="value-Repository"
        name="value-Repository"
        type="text"
      />
      <button
        class="w-full"
      >
        Search Repository
      </button>
    </section>
  </div>
</DocumentFragment>
`;
